components/brave_shields/core/browser/ad_block_component_service_manager.cc

```diff
@@ -342,6 +349,14 @@ void AdBlockComponentServiceManager::OnFilterListCatalogLoaded(
     const std::string& catalog_json) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   SetFilterListCatalog(FilterListCatalogFromJSON(catalog_json));
+
+  update_check_timer_.Start(
+      FROM_HERE, base::Minutes(kComponentUpdateCheckIntervalMins.Get()),
+      base::BindRepeating(
+          [](base::WeakPtr<AdBlockComponentServiceManager> component_manager) {
+            component_manager->UpdateFilterLists(base::DoNothing());
```

Comment:
this weak ptr requires a manual check as you're binding a lambda function, not a class method.

just add a class method to bind directly without fiddling with the weak ptr check. It's the safest and clear way of doing things.